### PR TITLE
Spi irq error test

### PR DESCRIPTION
**Pull request by engdoreis**
This PR adds a test checking the cmd busy error irq.

### CODE REVIEW

File: sw/device/tests/spi_host_irq_test.c [hunk 1]
```diff
@@ -291,6 +291,49 @@ static status_t rx_wm_event_irq(void) {
...
+  dif_spi_host_status_t status;
+
+  irq_fired = UINT32_MAX;
+  TRY(dif_spi_host_error_set_enabled(&spi_host, kErrorUnderTest, true));
+
+  TRY(dif_spi_host_get_status(&spi_host, &status));
+  TRY_CHECK(status.ready);
+  TRY_CHECK(!status.active);
+
+  // Overwhelm the cmd fifo to make the `STATUS.ready` goes low.
```
Comment (pamaury) [inline sw/device/tests/spi_host_irq_test.c:313]: nit: "go"

File: sw/device/tests/spi_host_irq_test.c [hunk 2]
```diff
@@ -291,6 +291,49 @@ static status_t rx_wm_event_irq(void) {
...
+    TRY(dif_spi_host_write_command(&spi_host, kDataSize / kCommands,
+                                   kDifSpiHostWidthStandard,
+                                   kDifSpiHostDirectionTx, true));
+  }
+
+  TRY(dif_spi_host_get_status(&spi_host, &status));
+  TRY_CHECK(!status.ready);
+  TRY_CHECK(status.active);
+
+  // // Wait for the error irq and check that it was triggered by
```
Comment (pamaury) [inline sw/device/tests/spi_host_irq_test.c:325]: double comment

**Review by pamaury — APPROVED** [review]
I am not an expert of the SPI host but it looks good to me overall.

File: sw/device/tests/spi_host_irq_test.c (hunk 1 above)
Comment (engdoreis) [reply sw/device/tests/spi_host_irq_test.c]: fixed.

File: sw/device/tests/spi_host_irq_test.c (hunk 2 above)
Comment (engdoreis) [reply sw/device/tests/spi_host_irq_test.c]: Fixed

**Review by cfrantz — APPROVED** [review]

File: sw/device/tests/spi_host_irq_test.c [hunk 3]
```diff
@@ -291,6 +291,49 @@ static status_t rx_wm_event_irq(void) {
...
+    kErrorUnderTest = kDifSpiHostErrorCmdBusy
+  };
+  static_assert(kDataSize % kCommands == 0, "Must be multiple.");
+
+  uint8_t data[kDataSize];
+  memset(data, 0xA5, kDataSize);
+  dif_spi_host_status_t status;
+
+  irq_fired = UINT32_MAX;
+  TRY(dif_spi_host_error_set_enabled(&spi_host, kErrorUnderTest, true));
```
[the remark this answers is not in the document]
Comment (engdoreis) [reply sw/device/tests/spi_host_irq_test.c]: Fixed.

**Pull request merged**